core/class/CommandLineTool.go

```diff
@@ -185,6 +190,41 @@ func (tool *CommandLineTool) placeInputFilesToCommandExecDir() error {
 	return nil
 }
 
+// defineStdoutDestination
+func (tool *CommandLineTool) defineStdinSource() error {
+	if tool.Root.Stdin != "" {
+		stdin, err := tool.Command.StdinPipe()
+		if err != nil {
+			return err
+		}
+		defer stdin.Close()
+		stdinfilename := tool.Root.Stdin
+		if strings.HasPrefix(stdinfilename, "$(") && strings.HasSuffix(stdinfilename, ")") {
+			// Create JavaScript runtime
+			vm, err := tool.Root.Inputs.ToJavaScriptVM(tool.Command.Dir)
+			if err != nil {
+				return err
+			}
+			retval, err := vm.Run(stdinfilename[2 : len(stdinfilename)-1])
+			if err != nil {
+				return err
+			}
+			stdinfilename = retval.String()
+		}
+		//
+		stdinfilepath := filepath.Join(tool.Command.Dir, stdinfilename)
+		stdinfile, err := os.Open(stdinfilepath)
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(stdin, bufio.NewReader(stdinfile))
```

Comment:
Do we need to wrap `*os.File` with `bufio.NewReader`? I guess we can pass `*os.File` directly to `io.Copy` because it implements `io.Reader` interface already.

Reply:
Thanks.
We do not need to create `bufio.NewReader` just pass the file.